src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -99,43 +99,60 @@ private ImmutableArray<ITypeSymbol> GetTypeSymbols(ImmutableArray<ISymbol> candi
             {
                 if (candidateSymbol is IMethodSymbol method)
                 {
-                    if (method.Parameters.Length > ordinalInInvocation)
+                    ITypeSymbol type;
+                    if (method.IsParams() && (ordinalInInvocation >= method.Parameters.Length - 1))
                     {
-                        var type = method.Parameters[ordinalInInvocation].Type;
-                        // If type is <see cref="Expression{TDelegate}"/>, ignore <see cref="Expression"/> and use TDelegate.
-                        // Ignore this check if expressionSymbol is null, e.g. semantic model is broken or incomplete or if the framework does not contain <see cref="Expression"/>.
-                        if (expressionSymbol != null &&
-                            type is INamedTypeSymbol expressionSymbolNamedTypeCandidate &&
-                            expressionSymbolNamedTypeCandidate.OriginalDefinition.Equals(expressionSymbol))
+                        if (method.Parameters.LastOrDefault()?.Type is IArrayTypeSymbol arrayType)
                         {
-                            var allTypeArguments = type.GetAllTypeArguments();
-                            if (allTypeArguments.Length != 1)
-                            {
-                                continue;
-                            }
+                            type = arrayType.ElementType;
+                        }
+                        else
+                        {
+                            continue;
+                        }
+                    }
+                    else if (ordinalInInvocation < method.Parameters.Length)
+                    {
+                        type = method.Parameters[ordinalInInvocation].Type;
+                    }
+                    else
+                    {
+                        continue;
+                    }
 
-                            type = allTypeArguments[0];
+                    // If type is <see cref="Expression{TDelegate}"/>, ignore <see cref="Expression"/> and use TDelegate.
+                    // Ignore this check if expressionSymbol is null, e.g. semantic model is broken or incomplete or if the framework does not contain <see cref="Expression"/>.
+                    if (expressionSymbol != null &&
+                        type is INamedTypeSymbol expressionSymbolNamedTypeCandidate &&
+                        expressionSymbolNamedTypeCandidate.OriginalDefinition.Equals(expressionSymbol))
+                    {
+                        var allTypeArguments = type.GetAllTypeArguments();
+                        if (allTypeArguments.Length != 1)
+                        {
+                            continue;
+                        }
+
+                        type = allTypeArguments[0];
+                    }
+
+                    if (type.IsDelegateType())
+                    {
+                        var methods = type.GetMembers(WellKnownMemberNames.DelegateInvokeName);
```

Comment:
Should we also support `params` in the case of delegates?

Reply:
Thank you, @JoeRobich !

Let us consider the following delegate:
```
        public delegate void Delegate1(Guid g, params Uri[] u);
        public void M(Delegate1 d) { }
```

1. `M((g, u) => {  u.$$ })` - here the compiler would consider `u` as an array not as the first element.
2. `M((g, u, v) => {  $$ })` - the delegate does not take 3 arguments.
3. `M((Guid g, params Uri[] u) => {  })` - params is not valid in the context.
4.  `M((Guid g, Uri[] u) => { u.$$ })` - here the compiler would consider `u` as an array not as the first element.

If you see an example, please let me know. Then, we will make a separate fix for it.